onnxruntime/core/providers/nnapi/nnapi_builtin/builders/helper.cc

```diff
@@ -264,6 +268,23 @@ bool HasValidQuantizationZeroPoints(const InitializedTensorSet& initializers, co
   return true;
 }
 
+float GetQuantizationScale(const InitializedTensorSet& initializers, const Node& node, size_t idx) {
+  const auto& scale_tensor = *initializers.at(node.InputDefs()[idx]->Name());
+  return GetTensorFloatData(scale_tensor)[0];
+}
+
+common::Status GetQuantizationZeroPoint(const InitializedTensorSet& initializers,
+                                        const Node& node, size_t idx, int32_t& zero_point) {
+  std::unique_ptr<uint8_t[]> unpacked_tensor;
```

Comment:
Could these have more generic names? Looks like one is reading a scalar float initializer (that it contains the scale is just a use case), an the other a scalar int32_t initializer.

Why do we use GetTensorFloatData for the float but manually unpack for the int32_t? Should we be doing the same for both, assumably the unpack so external data (although insane given it's a scalar initializer) is supported.

Reply:
The onnx zero point is uint8/int8, which I would have to use unpack (should be able to directly access the data pointer, but I would prefer to use unpack since only 1 int8 is copied), NNAPI is using int32_t,

Added comments about this